tests/setup_test.go

```diff
@@ -6517,3 +6539,206 @@ func TestProjectRunActions(t *testing.T) {
+	}
+	user02Token, _, err := gwUserAdminClient.CreateUserToken(ctx, agolaUser02, &gwapitypes.CreateUserTokenRequest{TokenName: "token01"})
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	gwUser02Client := gwclient.NewClient(sc.config.Gateway.APIExposedURL, user02Token.Token)
+
+	giteaAPIURL := fmt.Sprintf("http://%s:%s", sc.gitea.HTTPListenAddress, sc.gitea.HTTPPort)
+
+	giteaClient, err := gitea.NewClient(giteaAPIURL, gitea.SetToken(giteaToken))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	giteaRepo, project := createProject(ctx, t, giteaClient, gwUser01Client, withVisibility(gwapitypes.VisibilityPrivate))
+
+	push(t, config, giteaRepo.CloneURL, giteaToken, "commit", false)
+
+	_ = testutil.Wait(30*time.Second, func() (bool, error) {
+		runs, _, err := gwUser01Client.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+		if err != nil {
+			return false, nil
+		}
+
+		if len(runs) == 0 {
+			return false, nil
+		}
+		if runs[0].Phase != rstypes.RunPhaseFinished {
+			return false, nil
+		}
+
+		return true, nil
+	})
+
+	runs, _, err := gwUser01Client.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	if len(runs) == 0 {
+		t.Fatalf("expected %d run got: %d", 1, len(runs))
+	}
+
+	if runs[0].Phase != rstypes.RunPhaseFinished {
+		t.Fatalf("expected run phase %q, got %q", rstypes.RunPhaseFinished, runs[0].Phase)
+	}
+	if runs[0].Result != rstypes.RunResultSuccess {
+		t.Fatalf("expected run result %q, got %q", rstypes.RunResultSuccess, runs[0].Result)
+	}
+
+	_ = testutil.Wait(30*time.Second, func() (bool, error) {
+		runWebhookDeliveries, _, err := gwUser01Client.GetProjectRunWebhookDeliveries(ctx, project.ID, nil, &gwclient.ListOptions{Limit: 0, SortDirection: gwapitypes.SortDirectionAsc})
+		if err != nil {
+			return false, nil
+		}
+
+		if len(runWebhookDeliveries) != 4 {
+			return false, nil
+		}
+		for _, r := range runWebhookDeliveries {
+			if r.DeliveryStatus != gwapitypes.DeliveryStatusDelivered {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+
+	t.Run("test get project run webhook deliveries", func(t *testing.T) {
+		runWebhookDeliveries, resp, err := gwUser01Client.GetProjectRunWebhookDeliveries(ctx, project.ID, nil, &gwclient.ListOptions{Limit: 0, SortDirection: gwapitypes.SortDirectionAsc})
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+		if len(runWebhookDeliveries) != 4 {
+			t.Fatalf("expected 4 runWebhookDeliveries got: %d", len(runWebhookDeliveries))
+		}
+		if resp.Cursor != "" {
+			t.Fatalf("expected empty cursor, got %s", resp.Cursor)
+		}
+		for _, r := range runWebhookDeliveries {
+			if r.DeliveryStatus != gwapitypes.DeliveryStatusDelivered {
+				t.Fatalf("expected DeliveryStatus delivered, got %s", r.DeliveryStatus)
+			}
+		}
+	})
+
+	t.Run("test get project run webhook deliveries with limit less than project run webhook deliveries continuation", func(t *testing.T) {
```

Comment:
You aren't testing continuation. Since you have created only one delivery. You have to do the same as of other tests and use the same limit for easier test management.